Add LDAP user synchronization job

oliverguenther opened the pull request:
[OP#41677](https://community.openproject.com/wp/41677)
[OP#41655](https://community.openproject.com/wp/41655)

oliverguenther committed:
Add LDAP user synchronization job

machisuji commented on app/services/ldap/synchronize_users_service.rb:
```diff
@@ -0,0 +1,95 @@
+      login_filter = Net::LDAP::Filter.eq(ldap.attr_login, login)
+      ldap_con.search(base: base_dn,
+                      filter: ldap.default_filter & login_filter,
+                      attributes: search_attributes) do |entry|
```
@oliverguenther While you're at it, IMHO it would be nice to put this into a method like this:

```
def find_entries_by(login:)
  ldap_con.search(
    base: ldap.base_dn,
    filter: ldap.default_filter & login_filter(login), # making login_filter a method too
    attributes: search_attributes # making this a method too
  )
end
```

If only to make this easier to debug when I'm trying to find out why a user's LDAP config is wrong again.
When this is all in methods I just have to instantiate the service and override, for instance, the `login_filter` method to try out the different filters like this on the console:

```
service = ...
def service.login_filter(login); Net::LDAP::Filter.eq("foo", "bar"); end
```

Plus, this would also stop rubocop from complaining about the abc size!

oliverguenther committed:
Readd the two removed rake tasks as services

They were broken so I removed them. But instead of removing import
functionality, I decided to re-add them and fix them.

machisuji reviewed (approved):
Good stuff! 👍
